src/packet/buffer_tx.rs

```diff
@@ -309,6 +315,34 @@ impl PacketizingBuffer {
     pub fn ssrc(&self) -> Ssrc {
         self.ssrc.expect("Send buffer to have an SSRC")
     }
+
+    fn free_over_max_duration(&mut self, now: Instant) {
+        let Some(last) = self.queue.last() else { return };
+
+        let Some(last_ident) = self.queue.last_ident() else { return };
+        let Some(mut ident) = self.queue.first_ident() else { return };
+        let recent = last.meta.rtp_time.as_seconds();
+
+        loop {
+            if ident == last_ident {
+                break;
+            }
+
+            let Some(packet) = self.queue.get(ident) else { continue};
+            let packet_time = packet.meta.rtp_time.as_seconds();
```

Comment:
Don't you want to use the queue_time and not the rtp_time?  If you're writing and SFU, the rtp_time is time set by the original client, and you don't have any idea how that relates to your server's clock.

Reply:
We have no idea how RTP relates to the server clock, but since we only care about the `elapsed delta` it is "fine". With this approach time is driven forward by the arrival of new packets, so you get the effect of potentially holding for longer if no new data arrives. I do not see something "wrong" though.